src/openrct2/ReplayManager.cpp

```diff
@@ -0,0 +1,769 @@
+
+            uint32_t countChecksums = (uint32_t)data.checksums.size();
+            serialiser << countChecksums;
+
+            if (serialiser.IsLoading())
+            {
+                data.checksums.resize(countChecksums);
+            }
+
+            for (uint32_t i = 0; i < countChecksums; i++)
+            {
+                serialiser << data.checksums[i].first;
+                serialiser << data.checksums[i].second.raw;
+            }
+
+            return true;
+        }
+
+#ifndef DISABLE_NETWORK
+        void CheckState()
+        {
+            uint32_t checksumIndex = _currentReplay->checksumIndex;
+
+            if (checksumIndex >= _currentReplay->checksums.size())
+                return;
+
+            const auto& savedChecksum = _currentReplay->checksums[checksumIndex];
+            if (_currentReplay->checksums[checksumIndex].first == gCurrentTicks)
+            {
+                rct_sprite_checksum checksum = sprite_checksum();
+                if (savedChecksum.second.ToString() != checksum.ToString())
+                {
+                    // Detected different game state.
+                    log_verbose(
+                        "Different sprite checksum at tick %u ; Saved: %s, Current: %s", gCurrentTicks,
+                        savedChecksum.second.ToString().c_str(), checksum.ToString().c_str());
+
+                    _faultyChecksumIndex = checksumIndex;
+                }
+                else
+                {
+                    // Good state.
+                    log_verbose(
+                        "Good state at tick %u ; Saved: %s, Current: %s", gCurrentTicks,
+                        savedChecksum.second.ToString().c_str(), checksum.ToString().c_str());
+                }
+                _currentReplay->checksumIndex++;
+            }
+        }
+#endif // DISABLE_NETWORK
+
+        void ReplayCommands()
+        {
+            auto& replayQueue = _currentReplay->commands;
+
+            while (replayQueue.begin() != replayQueue.end())
+            {
+                const ReplayCommand& command = (*replayQueue.begin());
+
+                if (_mode == ReplayMode::PLAYING)
+                {
+                    // If this is a normal playback wait for the correct tick.
+                    if (command.tick != gCurrentTicks)
+                        break;
+                }
+                else if (_mode == ReplayMode::NORMALISATION)
+                {
+                    // Allow one entry per tick.
+                    if (gCurrentTicks != _nextReplayTick)
+                        break;
+
+                    _nextReplayTick = gCurrentTicks + 1;
+                }
+
+                bool isPositionValid = false;
+
+                if (command.action != nullptr)
+                {
+                    GameAction* action = command.action.get();
+                    action->SetFlags(action->GetFlags() | GAME_COMMAND_FLAG_REPLAY);
+
+                    Guard::Assert(action != nullptr);
```

Comment:
This assert doesn't matter, you've already dereferenced it above. Reorder.

Reply:
And does it even make sense? You've already checked this condition in line 718, the condition for this block.

Reply:
I suppose not, debugging left overs.